src/Workspaces/CSharp/Portable/Extensions/CastExpressionSyntaxExtensions.cs

```diff
@@ -177,25 +176,44 @@ private static bool UserDefinedConversionIsAllowed(ExpressionSyntax expression,
                 }
 
                 var parameter = argument.DetermineParameter(semanticModel, cancellationToken: cancellationToken);
-                if (parameter != null && parameter.IsParams)
+                return CheckConversionOfParamToParamArray(parameter, castType, semanticModel);
+            }
+
+            if (parent is AttributeArgumentSyntax attributeArgument)
+            {
+                if (attributeArgument.Parent is AttributeArgumentListSyntax attributeArgumentList)
                 {
-                    Debug.Assert(parameter.Type is IArrayTypeSymbol);
+                    // We don't check the position of the argument because in attributes it is allowed that 
+                    // params parameter are positioned in between if named arguments are used.
+                    // The *single* argument check above is also broken: https://github.com/dotnet/roslyn/issues/20742
+                    var parameter = attributeArgument.DetermineParameter(semanticModel, cancellationToken: cancellationToken);
+                    return CheckConversionOfParamToParamArray(parameter, castType, semanticModel);
+                }
+            }
 
-                    var parameterType = (IArrayTypeSymbol)parameter.Type;
+            return false;
+        }
 
-                    var conversion = semanticModel.Compilation.ClassifyConversion(castType, parameterType);
-                    if (conversion.Exists &&
-                        conversion.IsImplicit)
-                    {
-                        return false;
-                    }
+        private static bool CheckConversionOfParamToParamArray(IParameterSymbol parameter, ITypeSymbol castType, SemanticModel semanticModel)
+        {
+            if (parameter != null && parameter.IsParams)
+            {
+                Debug.Assert(parameter.Type is IArrayTypeSymbol);
 
-                    var conversionElementType = semanticModel.Compilation.ClassifyConversion(castType, parameterType.ElementType);
-                    if (conversionElementType.Exists &&
-                        conversionElementType.IsImplicit)
-                    {
-                        return true;
-                    }
+                var parameterType = (IArrayTypeSymbol)parameter.Type;
```

Comment:
Is this correct in broken code?  What if I type a method that says `M(params int something)` Does the symbol still report `IsParams`?